File: implement/p60057/Solution.java
package p60057;

public class Solution {
    public static void main(String[] args) {
        int result = solve("aabbaccc");
        System.out.println(result);

    }

    static int solve(String s) {
        int minLen = s.length();

        for (int i = 1; i < s.length() / 2 + 1; i++) {
            StringBuffer newStr = new StringBuffer();
            int repeatCnt = 1;

            for (int j = 0; j < s.length(); j = j + i) {
                String a;
                if (j + i < s.length()) {
                    a = s.substring(j, j + i);
                } else {
                    a = s.substring(j);
                }

                if (a.length() != i) {
                    if (repeatCnt == 1) {
                        newStr.append(a);
                    } else {
                        newStr.append(String.valueOf(repeatCnt) + a);
                    }
                    repeatCnt = 1;
                    break;
                }

                String b;
                if (j + i + i < s.length() && j != s.length() - 1) {
                    b = s.substring(j + i, j + i + i);
                } else {
                    b = s.substring(j + i);
                }

                if (a.equals(b)) {
                    repeatCnt++;
                } else {
                    if (repeatCnt == 1) {
                        newStr.append(a);
                    } else {
                        newStr.append(String.valueOf(repeatCnt) + a);
                    }
                    repeatCnt = 1;
                }
            }

            if (newStr.length() < minLen) {
                minLen = newStr.length();
            }
        }

        return minLen;
    }

}

File: implement/p60057/solution.py
def solution(s):
    min_len = len(s)  # 8

    for i in range(1, len(s) // 2 + 1):  # 1,2,3,4
        result_str = ''
        repeat = 1
        for start_idx in range(0, len(s), i):  # i==1: 0~7, i==2: 0~6(0,2,4,6),
            a = s[start_idx: start_idx + i]
            b = s[start_idx + i: start_idx + (i * 2)]
            if a == b:
                repeat += 1
            else:
                if repeat == 1:
                    result_str += a
                else:
                    result_str += str(repeat) + a

                repeat = 1  # reset

        if len(result_str) < min_len:
            min_len = len(result_str)

    return min_len


result = solution("aabbaccc")
print(result)  # 7
result = solution("abcabcabcabcdededededede")
print(result)  # 14
